src/Apps/Order2/Routes/Checkout/Hooks/useLoadCheckout.ts

```diff
@@ -0,0 +1,104 @@
+import createLogger from "Utils/logger"
+import type { useLoadCheckout_order$key } from "__generated__/useLoadCheckout_order.graphql"
+import { every } from "lodash"
+import { useCallback, useEffect, useState } from "react"
+import { useFragment } from "react-relay"
+import { graphql } from "react-relay"
+
+const logger = createLogger("useLoadCheckout")
+
+const MINIMUM_LOADING_MS = 1000
+
+type CheckoutLoadingError = "missing_line_item_data"
+
+export const useLoadCheckout = (order?: useLoadCheckout_order$key | null) => {
+  const orderData = useFragment(FRAGMENT, order)
+  const [isLoading, setIsLoading] = useState(true)
+  const [loadingError, setLoadingError] = useState<CheckoutLoadingError | null>(
+    null,
+  )
+  const [minimumLoadingPassed, setMinimumLoadingPassed] = useState(false)
+  const [expressCheckoutLoaded, setExpressCheckoutLoaded] = useState(false)
+  const [orderValidated, setOrderValidated] = useState(false)
+
+  const checks = [minimumLoadingPassed, orderValidated, expressCheckoutLoaded]
+
+  const handleExpressCheckoutLoaded = useCallback(() => {
+    if (expressCheckoutLoaded) {
+      return
+    }
+    setExpressCheckoutLoaded(true)
+  }, [expressCheckoutLoaded])
```

Comment:
Callback to use in the express checkout element after it loads. All of this stuff could be promoted to a context/dispatch system for the checkout flow as well.